### ISSUE TITLE
Saving invoice with line items

### ISSUE THREAD

**Issue by jasonlewis** [calcinai/xero-php#394]
When saving a new invoice with line items added via `addLineItem` the response from Xero includes the `LineItemID` for each of the items but due to the nature of the save this is never updated on the `LineItem` model.

This stems from the fact that the `save` method on a model seems to not allow the data to be replaced. That is, on the application, you can call `save` with the `$replace_data` parameter set to `true`, this then let's the package update the local model with the fresh data from Xero.

Does this sound like something that could be tweaked? Perhaps allowing the shorthand call to `save` from a model accept the replacing data parameter.

**Comment by calcinai** [comment]
It's funny you mentioned this, because I thought I fixed this this other day.  I feel like this soul be the default behaviour, no idea why it isn't..

**Comment by jasonlewis** [comment]
Yeah I agree, I was slightly confused at first when I wasn't seeing the underlying models updating.

Obviously this affects more than just line items for an invoice as it would also mean every property on a model will be updated with the Xero response. For my use case that makes sense and is what I'd expect, but I can't say everyone would expect that.

Perhaps add the option to replace the data on the shorthand save, default it to `true`, but it can obviously be set to `false` to maintain the previous behavior.

**Comment by calcinai** [comment]
That sounds reasonable to me

**Comment by joelgomezb** [comment]
$xero = \App::make('XeroPrivate');
          $invoice = \App::make('XeroInvoice');
          $contacts = \App::make('XeroContact');
          $line1 = \App::make('XeroInvoiceLine');
         ........
          $line1->setDescription('Attention: ' . $contact[0]->Name . '|' . $request->reason);
          $line1->setQuantity(1);
          $line1->setLineItemID($this->getGUID());
          $line1->setAccountCode($request->account);
          $line1->setUnitAmount($request->amount);
          $line1->setTaxAmount(0);
          $line1->setLineAmount($request->amount);
          // $line1->setDiscountRate(0); // Percentage

          // Add the line to the order
          $invoice->addLineItem($line1);

         $xero->save($invoice, true);

The lines dont save... can i help me?
chears

**Comment by joelgomezb** [comment]
Thks

**Comment by moshahed** [comment]
Hi  @pineoclean  @joelgomezb @calcinai 
I am facing the similar issue. Can you explain it little bit more, how can I make it work?
Where should I put `$this->propertyUpdated('LineItems', $value);`

My Code Sample in below. Currently only working creating new line item but can't update existing line items.
```
$xero = $this->getXeroConnection();
$invoice = $xero->loadByGUID('Accounting\\Invoice', 'invoice-id');

$lineitem = new \XeroPHP\Models\Accounting\Invoice\LineItem($xero);
$lineitem->setDescription("Demo Desc")
               ->setGUID('demo-line-item');
$invoice->addLineItem($lineitem); 
$xero->save($invoice, true);
```
Thanks in advance
